fish-rust/src/common.rs

```diff
@@ -1649,11 +1662,11 @@ static IS_WINDOWS_SUBSYSTEM_FOR_LINUX: Lazy<bool> = Lazy::new(|| {
     let dash = info.release.iter().position('-');
 
     if dash
-        .map(|d| unsafe { libc::strtod(std::ptr::addr_of!(info.release[d + 1]), std::ptr::null()) } >= 17763)
+        .map(|d| unsafe { libc::strtod(&info.release[d + 1], std::ptr::null()) } >= 17763)
```

Comment:
Could this use [wcstoi](https://github.com/fish-shell/fish-shell/blob/master/fish-rust/src/wutil/wcstoi.rs) instead? Version numbers aren't floats.